packages/loader/container-loader/src/container.ts

```diff
@@ -1237,8 +1250,13 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 if (key === "code" || key === "code2") {
                     debug(`loadRuntimeFactory ${JSON.stringify(value)}`);
 
-                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                    this.reloadContext();
+                    if (shouldRefresh) {
```

Comment:
If I got it right, shouldRefresh === true means we expect host to reload page / iFrame / etc.
Question on both branches of that IF: do we assume no local state?
If there are any pending changes, what is expected? Runtime today is not capable of summarizing in presence of local state. I assume we rely on "simple" summary to be safe in presence of local state as it uses committed ops, right?
It would be great to use somewhat strong lingvo across layers, such that it's obvious that this kind of summary is special and has to work in such conditions.

Reply:
Right now this is why we _must_ use differential summary like you said.
Not sure how to clarify this other than maybe add some comments where it is used?

Reply:
this shouldn't go here. probably in the reloadContext method. my latest PR might make this clearer

Reply:
Removed shouldRefresh per discussion.  Now it will always race to summarize.